### DISCUSSION TITLE
disable date interpolation when grouping by date

### DISCUSSION THREAD

**Discussion by GREEB** [perspective-dev/perspective#1970] [Q&A] [answered]
I'm not sure how to call this but when grouping by date perspective adds dates that may not be in the data, i assume this is for the zooming to work well. If you have financial data that never has weekend data this creates holes as expected.

Is there a way to keep the zoom but disable creation of new dates?
Whats the quickest way to transform date to string? this would disable the date zooming and not create new dates, but still display a label. _i had some problems when creating a new column with `string("date")` my python streamer crashes_

![image](https://user-images.githubusercontent.com/1221769/190866150-d8d82c83-7507-4b46-9c96-4b1372504240.png)
**here you see the gaps**

Thanks for any help

**Comment by texodus** [comment, marked as answer]
>  i had some problems when creating a new column with string("date") my python streamer crashes

Need a repro please - this is the correct solution.  `date` axes are linear, and adding "business time" awareness is very complicated.

**Comment by GREEB** [comment]
Because our streamer runs on a VPS it ran out of [resources](https://pcr.cloud-mercato.com/providers/hetzner/flavors/cx41) and got killed, sorry bout that. Thanks for confirming.

We are trying to serve 1.5GB (6M rows), testing with 1M rows works much better, are there any recommendations for bigger data-sets?  We'll try [multiprocessing](https://github.com/finos/perspective/discussions/1588).

Thanks for all the help

**Reply by texodus** [reply]
Definitely not confirming, I misunderstood - the way you stated the problem originally, implied the expression `date("column")` was _working_ in WebAssembly and failing in Python.  This is not the case though - the solution to your _actual_ issue is to upgrade your VPS.

`multiprocessing` will absolutely make any performance issues you have _substantially_ worse.

**Reply by GREEB** [reply]
was talking about confirming that `string` is the right way to convert date to string to break linearity.

> `multiprocessing` will absolutely make any performance issues you have _substantially_ worse.

good to know, thanks.
